uberfire-extensions/uberfire-metadata/uberfire-metadata-backends/uberfire-metadata-backend-infinispan/src/main/java/org/uberfire/ext/metadata/backend/infinispan/provider/MappingProvider.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.uberfire.ext.metadata.backend.infinispan.provider;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.uberfire.ext.metadata.analyzer.ElasticSearchAnalyzerWrapper;
+import org.uberfire.ext.metadata.backend.infinispan.proto.schema.Field;
+import org.uberfire.ext.metadata.backend.infinispan.proto.schema.Message;
+import org.uberfire.ext.metadata.backend.infinispan.proto.schema.ProtobufScope;
+import org.uberfire.ext.metadata.backend.infinispan.proto.schema.ProtobufType;
+import org.uberfire.ext.metadata.backend.infinispan.proto.schema.Schema;
+import org.uberfire.ext.metadata.backend.infinispan.utils.AttributesUtil;
+import org.uberfire.ext.metadata.model.KObject;
+import org.uberfire.ext.metadata.model.KProperty;
+import org.uberfire.ext.metadata.model.impl.KPropertyImpl;
+import org.uberfire.java.nio.base.version.VersionHistory;
+import org.uberfire.java.nio.file.attribute.FileTime;
+
+public class MappingProvider {
+
+    public static final String CHECKIN_COMMENT = "checkinComment";
```

Comment:
Are these the same field names that are declared as constants in `MappingFieldFactory` and in `KObjectMarshaller` too? I think you should declare these constants in one place (e.g. something like InfinispanFieldNames) and reuse those definitions in all 3 places.

Reply:
In the only place I didn't change anything is in MappingFieldFactory because is part ES and I think we will remove those bytes, right?